workstreams/events_in_cicd/vocabulary.md

```diff
@@ -0,0 +1,108 @@
+- **Repository Created Event**: a new Source Code Repository was created to host source code for a project
+- **Repository Modified Event**: a Source Code Repository modified some of its attributes, like location, or owner
+- **Repository Deleted Event**: a Source Code Repository was deleted and it is not longer available
+- **Branch Created Event**: a Branch inside the Repository was created 
+- **Branch Deleted Event**: a Branch inside the Repository was deleted
+- **Change Committed**: a change is commited to a branch
+- **Change Approved**: a change request is created by a user, that requires review and approval before being applied to an existing branch inside the repository
+- **Change Rejected**: a change was reviewed and rejected by a user
+- **Change Merged**: a change request has been merged to a branch in an existing repository
+
+
+Repository Events MUST include the following attributes:
+- **Event Type**: the type is restricted to include `CD.Repository**` prefix. For example `CD.Repository.Created` or `CD.Repository.ChangeApproved`
+- **Repository URL**: indicates the location of the source code repository
+- **Repository Name**: friendly name to list this repository to users
+
+Optional attributes: 
+- **Repository Owner**: indicates who is the owner of the repository, usually a `user` or an `organization`
+
+
+# Continuous Integration Pipeline Events
+
+These events are related to continuous integration pipelines, this pipelines usually include building, testing, packaging and releasing software artifacts. 
+Due the dynamic nature of Pipelines, most of actual work needs to be queued to happen in a distributed way, hence Queued events are added. 
+Adopters can choose to ignore these events if they don't apply to their use cases. 
+
+A pipeline, in the context of Continuous Integration, is the definition of a set of tasks that needs to be performed to build, test, package and release software artifacts. A pipeline can be instanciated multiple times, for example to build different versions of the same artifact. That instance will have a unique Id and it will help us to track the build and release progress on a particular software artifact. 
+
+- **PipelineInstance Queued**: a PipelineInstance has been schedule to run
+- **PipelineInstance Started**: a PipelineInstance has started and it is running
+- **PipelineInstance Finished**: a Pipeline has finished it execution, the event will contain the finished status, success, error or failure
+- **Build Queued**: a Build task has been queued, this build process usually is in charge of producing a binary from source code
+- **Build Started**: a Build task has started 
+- **Build Finished**: a Build task has finished, the event will contain the finished status, success, error or failure
+- **Tests Started**: a Test task has started
+- **Tests Finished**: a Test task has finished, the event will contain the finished status, success, error or failure
```

Comment:
Would tasks such as static code analysis etc. be considered tests?

Also, could tests be queued as well?

Reply:
@erkist very vaild point.. let me reflect that.. We need a category for things like code analysis.. and security scanning.. any suggestions? We need broader categories that can serve as boundaries.. then we can go into the concrete ones.. or even allow people to extend this spec with their own specific events.

Reply:
We could consider adding a generic "activity" entity with queued/started/finished, but then Build and Tests may both be considered activities.